### PR TITLE
Update tool and plugins version (FT-5906)

### PR DESCRIPTION
**Pull request by DReigada**

### CODE REVIEW

**Comment by DReigada** [comment]
Also closes #17

File: src/main/scala/codacy/codesniffer/docsgen/GeneratorMain.scala [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package codacy.codesniffer.docsgen
+
+object GeneratorMain extends App {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/GeneratorMain.scala]: Please don't use app it has bugs

File: src/main/scala/codacy/codesniffer/docsgen/Generator.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import codacy.codesniffer.docsgen.parsers._
+import com.codacy.plugins.api.results.Tool
+import play.api.libs.json.{Json, Writes}
+
+import scala.collection.parallel.ForkJoinTaskSupport
+import scala.collection.parallel.immutable.ParSeq
+
+class Generator() {
+
+  private[this] val toolName = Tool.Name("PHP Code Sniffer")
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/Generator.scala]: This should be the same that we have as shortName inside codacy-plugins.
This should be a string without spaces and usually simple as an ID.

File: src/main/scala/codacy/codesniffer/docsgen/Generator.scala [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  private[this] val toolVersion = None
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/Generator.scala]: Add version:
* Read from a files in the home of the projects and also use it for docker build
* When generating write it in patterns.json and then when building docker read from patterns.json

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+  def patterns: Set[PatternDocs] =
+    withRepo(repositoryURL)(handleRepo)
+      .fold(a => throw a, identity)
+
+  private[this] def withRepo[A](repositoryURL: String)(f: File => A): Either[Throwable, A] = {
+    val dir = Files.createTempDirectory("")
+    for {
+      _ <- CommandRunner
+        .exec(List("git", "clone", repositoryURL, dir.toString))
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Always clone a specific version matching the one used to build the docker.

File: src/main/scala/codacy/codesniffer/docsgen/Generator.scala (hunk 2 above)
Comment (DReigada) [reply src/main/scala/codacy/codesniffer/docsgen/Generator.scala]: This was the name that was defined before

File: src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala (hunk 4 above)
Comment (DReigada) [reply src/main/scala/codacy/codesniffer/docsgen/parsers/DocsParser.scala]: Fixed

File: src/main/scala/codacy/codesniffer/docsgen/parsers/MagentoCSDocsParser.scala [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+class MagentoCSDocsParser extends DocsParser {
+
+  override val repositoryURL = "https://github.com/magento/marketplace-eqp.git"
+
+  override def checkoutCommit: String = VersionsHelper.magento
+
+  private[this] val sniffRegex = """.*(MEQP1|MEQP2)\/Sniffs\/(.*?)\/(.*?)Sniff.php""".r
+
+  def handleRepo(dir: File): Set[PatternDocs] = {
+    (for {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/parsers/MagentoCSDocsParser.scala]: Use map with collection breakout

Comment (DReigada) [reply src/main/scala/codacy/codesniffer/docsgen/parsers/MagentoCSDocsParser.scala]: Fixed

File: src/main/scala/codacy/codesniffer/docsgen/VersionsHelper.scala [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  lazy val codesniffer = properties.getProperty("php-codesniffer")
+  lazy val magento = properties.getProperty("magento-eqp")
+  lazy val wordpress = properties.getProperty("wordpress")
+  lazy val phpCompatibility = properties.getProperty("php-compatibility")
+
+}
+
+object Test extends App {
```
Comment (rtfpessoa) [inline src/main/scala/codacy/codesniffer/docsgen/VersionsHelper.scala]: Do we need this? If yes can we get a better name?

Comment (DReigada) [reply src/main/scala/codacy/codesniffer/docsgen/VersionsHelper.scala]: We don't, I just forgot to delete it 🙃

**Pull request merged**